pkg/cli/mirror/cincinnati_graph_image.go

```diff
@@ -167,3 +177,55 @@ func downloadGraphData(ctx context.Context, dir string) error {
 	_, err = io.Copy(out, resp.Body)
 	return err
 }
+
+// extracts the tarball in target directory.
+func untar(tarball, target string) error {
+	reader, err := os.Open(tarball)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	uncompressedStream, err := gzip.NewReader(reader)
+	if err != nil {
+		return fmt.Errorf("failed to read uncompressed stream of tarball: %v", err)
+	}
+
+	tarReader := tar.NewReader(uncompressedStream)
+
+	for true {
+		header, err := tarReader.Next()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return fmt.Errorf("failed to iterate through tarball: %v", err)
+		}
+		fPath := filepath.Join(target, header.Name)
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(fPath, 0755); err != nil {
+				return fmt.Errorf("failed to create directories while extracting tar: %v", err)
+			}
+		case tar.TypeReg:
+			outFile, err := os.Create(fPath)
+			if err != nil {
+				return fmt.Errorf("failed to create file while extracting tar: %v", err)
+			}
+			if _, err := io.Copy(outFile, tarReader); err != nil {
+				return fmt.Errorf("failed to copy extracted file: %v", err)
+			}
+			outFile.Close()
```

Comment:
technically this is a leak on errors, but I don't think it matters that much in `oc`

Reply:
using archiver library instead of this function